config/metricman.schema.exs

```diff
@@ -21,61 +21,64 @@
+                 []
+               [{_, tags}] ->
+                 tags = for tag <- tags do
+                   case String.split(tag, ":") do
+                     [key, value] -> {key |> String.to_atom, value}
+                     _ -> exit("Unsupported tags for InfluxDB: #{inspect tags}")
+
+                   end
+                 end
+                 [tags: tags]
             end
-            params = Access.get(acc || [], :exometer_report_influxdb, [])
-            Keyword.put(acc || [], :exometer_report_influxdb, params ++ [protocol: protocol |> String.to_atom, host: host, port: port] ++ db)
-          _ ->
-            exit("Unsupported URI for InfluxDB: #{db}")
-        end
-      _, db, _ ->
-        exit("Unsupported URI for InfluxDB: #{db}")
-    end,
-    "metrics.influx.tags": fn
-      _mapping, [""], acc -> acc
-      _mapping, _, nil -> nil
-      _mapping, tags, acc ->
-        tags = for tag <- tags do
-          case String.split(tag, ":") do
-            [key, value] -> {key |> String.to_atom, value}
-            _ ->
-              exit("Unsupported tags for InfluxDB: #{inspect tags}")
-          end
-        end
-        if Keyword.has_key?(acc, :exometer_report_influxdb) do
-          params = Access.get(acc, :exometer_report_influxdb, [])
-          Keyword.put(acc, :exometer_report_influxdb, params ++ [tags: tags])
-        else acc end
-      _, tags, _ ->
-        exit("Unsupported tags for InfluxDB: #{inspect tags}")
-    end,
-    "metrics.influx.batch_window_size": fn
-      _mapping, _, nil -> []
-      _mapping, window_size, acc ->
-        if not is_integer(window_size) do
-          exit("Unsupported batch_window_size for InfluxDB: #{inspect window_size}")
-        end
-        if Keyword.has_key?(acc, :exometer_report_influxdb) do
-          params = Access.get(acc, :exometer_report_influxdb)
-          Keyword.put(acc, :exometer_report_influxdb, params ++ [batch_window_size: window_size])
-        else acc end
-    end
+      batch_window_size = case Conform.Conf.get(table, "exometer_core.report.reporters.batch_window_size") do
+                            [{_, []}] ->
+                              []
+                            [{_, window_size}] ->
+                              if not is_integer(window_size) do
+                                exit("Unsupported batch_window_size for InfluxDB: #{inspect window_size}")
+                              end
+                             [batch_window_size: window_size]
+                          end
+      # Delete extra fields from the mappings, we no need in it
+      # anymore. In other way we will see it in sys.config
+      # TODO delete it with match
+      :ets.delete(table, ['exometer_core','report','reporters', 'db'])
+      :ets.delete(table, ['exometer_core','report', 'reporters', 'tags'])
+      :ets.delete(table, ['exometer_core','report','reporters', 'batch_window_size'])
+      [exometer_report_influxdb: db ++ tags ++ batch_window_size]
```

Comment:
I did not test it but I assume this will remove all previously proceeded reporter options. So let's say you have another reporter in you app config (and in its schema) then this config is completely overwritten by this line and will NOT be part of the final config.